modules/33acrossBidAdapter.js

```diff
@@ -194,18 +194,44 @@ function _createServerRequest({bidRequest, gdprConsent = {}, uspConsent, pageUrl
   // therefore in ad targetting process
   ttxRequest.id = bidRequest.bidId;
 
-  // Set GDPR related fields
-  ttxRequest.user = {
-    ext: {
-      consent: gdprConsent.consentString
-    }
-  };
-  ttxRequest.regs = {
-    ext: {
-      gdpr: (gdprConsent.gdprApplies === true) ? 1 : 0,
-      us_privacy: uspConsent || null
-    }
-  };
+  if (gdprConsent.consentString) {
+    ttxRequest.user = setExtension(
+      ttxRequest.user,
+      'consent',
+      gdprConsent.consentString
+    )
+  }
+
+  if (Array.isArray(bidRequest.userIdAsEids)) {
+    ttxRequest.user = setExtension(
+      ttxRequest.user,
+      'eids',
+      bidRequest.userIdAsEids
+    )
+  }
+
+  if (gdprConsent.gdprApplies === true) {
+    ttxRequest.regs = setExtension(
+      ttxRequest.regs,
+      'gdpr',
+      1
+    )
+  } else {
+    ttxRequest.regs = setExtension(
+      ttxRequest.regs,
+      'gdpr',
+      0
+    )
+  }
+
+  if (uspConsent) {
```

Comment:
Optional feedback...
I am seeing many `if` statements checking for the truthyness of the subfields

could it be possible to move that `if` statement to setExtension? and probably just keep the one that checks for isArray.

e.g.
```
function setExtension(obj = {}, key, value='') { // Values such as undefined will default to an empty string
  if (!String(value)) {  // Values such as 0 or null will be valid for the "ext" assignment
    return Object.assign({}, obj);
  }
  
  return Object.assign({}, obj, {
    ext: Object.assign({}, obj.ext, {
      [key]: value
    })
  });
}
```

Reply:
Can you elaborate why you are recommending this? In its current form the check to truthyness makes it very clear to the reader when the ORTB field ought to be set or not.

Reply:
Similar to the feedback I provided in the `gdprApplies` suggestion above, I was trying to avoid behavior conditional branching,  in my opinion the first `if (!String(value))` statement still makes it clear the check for truthyness, but maybe this solution is unnecessarily more elaborated... or maybe the function name `setExtension` is not ideal to make the point clear that the extension will be set based on a condition, it could be renamed to `setExtensionIfPresent`..

either way if you find it to be more confusing, feel free to ignore this suggestion.

Reply:
So I want to keep `setExtension` simple, it does what it's asked to do i.e. set extension.  The caller has the responsibility to figure out if that extension needs to be set or not. Setting an extension fields may not always predicate on the truthyness of a condition, we may want to set it regardless. So I'll keep this code structure as is for the present. Also, I like the fact that branching it makes it clear to the reader what the logic is and also to the developer to add add unit tests accordingly (to test for the outcomes of the condition or not)